pkg/store/bucket.go

```diff
@@ -863,14 +863,27 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 	req.MaxTime = s.limitMaxTime(req.MaxTime)
 
 	var (
-		ctx     = srv.Context()
-		stats   = &queryStats{}
-		res     []storepb.SeriesSet
-		mtx     sync.Mutex
-		g, gctx = errgroup.WithContext(ctx)
-		hints   = &hintspb.SeriesResponseHints{}
+		ctx              = srv.Context()
+		stats            = &queryStats{}
+		res              []storepb.SeriesSet
+		mtx              sync.Mutex
+		g, gctx          = errgroup.WithContext(ctx)
+		resHints         = &hintspb.SeriesResponseHints{}
+		reqHintsMatchers = []*labels.Matcher(nil)
 	)
 
+	if req.Hints != nil {
+		reqHints := &hintspb.SeriesRequestHints{}
+		if err := types.UnmarshalAny(req.Hints, reqHints); err != nil {
+			return status.Error(codes.InvalidArgument, errors.Wrap(err, "unmarshal series request hints").Error())
+		}
+
+		reqHintsMatchers, err = promclient.TranslateMatchers(reqHints.BlockMatchers)
```

Comment:
side note: we keep useful helpers in `storepb/custom.go` would be nice to move `TranslateMatchers` there as well at some point :thinking:

Reply:
Also can make sure those are block matchers (in name)
```suggestion
		reqBlockMatchers, err = promclient.TranslateMatchers(reqHints.BlockMatchers)
```